notebooks/fractal_persistence.ipynb

```diff
@@ -214,7 +216,102 @@
     "    plt.savefig(f\"../assets/disco{time_stamp}_{idx+2}.png\")\n",
     "    plt.show() \n",
     "       \n",
-    "    print(msg2 + msg)"
+    "    print(msg2 + msg)\n",
+    "    # save results\n",
+    "    # results_img, accumulated_t, total_steps, explode, vanish, done, grid_0, grid\n",
+    "    \n",
+    "    img_savepath = os.path.join(save_dir, f\"{exp_name}_img_{idx}.png\")\n",
+    "    img_npy_savepath = os.path.join(save_dir, f\"{exp_name}_img_{idx}.npy\")\n",
+    "    \n",
+    "    accumulated_t_savepath = os.path.join(save_dir, f\"{exp_name}_accumulated_t_{idx}.npy\")\n",
+    "    total_steps_savepath = os.path.join(save_dir, f\"{exp_name}_total_steps_{idx}.npy\")\n",
+    "    \n",
+    "    explode_savepath = os.path.join(save_dir, f\"{exp_name}_explode_{idx}.npy\")\n",
+    "    vanish_savepath = os.path.join(save_dir, f\"{exp_name}_vanish_{idx}.npy\")\n",
+    "    done_savepath = os.path.join(save_dir, f\"{exp_name}_done_{idx}.npy\")\n",
+    "    \n",
+    "    grid_0_savepath = os.path.join(save_dir, f\"{exp_name}_grid_0_{idx}.npy\")\n",
+    "    grid_T_savepath = os.path.join(save_dir, f\"{exp_name}_grid_T_{idx}.npy\")\n",
+    "                                    \n",
+    "    sio.imsave(img_savepath, results[-1][0])\n",
+    "    np.save(img_npy_savepath, results[-1][0])\n",
+    "    np.save(accumulated_t_savepath, results[-1][1])\n",
+    "    np.save(total_steps_savepath, results[-1][2])\n",
+    "    np.save(explode_savepath, results[-1][3])\n",
+    "    np.save(vanish_savepath, results[-1][4])\n",
+    "    np.save(done_savepath, results[-1][5])\n",
+    "    np.save(grid_0_savepath, results[-1][6])\n",
+    "    np.save(grid_T_savepath, results[-1][7])\n",
+    "    \n",
+    "    # log experiment metadata\n",
+    "    #metadata = \"index, min_dt, max_dt, min_kr, max_kr, parameter_steps, time_stamp, \"\n",
+    "    #metadata += \"img_savepath, accumulated_t_savepath, total_steps_savepath, explode_savepath, vanish_savepath, grid_T_savepath\\n\"\n",
+    "\n",
+    "    metadata = f\"{idx}, {pattern_name}, {min_dt}, {max_dt}, {min_kr}, {max_kr}, {parameter_steps}, {time_stamp}, {t2-t1:2f}, {t2-t0:2f}, \"\n",
+    "    metadata += f\"{img_savepath}, {accumulated_t_savepath}, {total_steps_savepath}, {explode_savepath}, {vanish_savepath}, {grid_T_savepath}\\n\"\n",
+    "    with open(metadata_path,\"a\") as f:\n",
+    "        f.write(metadata)\n",
+    "        \n",
+    "    # determine next parameter range\n",
+    "    freq_zoom_dim = (results[-1][0].shape[-2]) // freq_zoom_fraction\n",
+    "    freq_zoom_stride = 4\n",
+    "    freq_zoom_strides = (results[-1][0].shape[-2]-freq_zoom_dim) // freq_zoom_stride +1\n",
+    "    \n",
+    "    fzd = freq_zoom_dim\n",
+    "    fzs = freq_zoom_stride\n",
+    "    \n",
+    "    params_list = []\n",
+    "    entropy = []\n",
+    "    frequency_entropy = []\n",
+    "    frequency_ratio = []\n",
+    "    # Weighted RGB conversion to grayscale\n",
```

Comment:
This emphasizes the green channel (conditions considered stable for the duration of up to max_t) while still considering the exploded (red) and vanished (blue) channels.

Actually, the red and blue weights here should be swapped.